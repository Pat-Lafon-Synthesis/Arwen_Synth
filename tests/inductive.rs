use arwen_synth::{
    language::{Constant, Examples, TestCase},
    synthesis,
    types::BaseType,
};

mod libraries;
use libraries::*;

#[test]
fn inductive_synthesis_negation() {
    let sig = arwen_synth::types::Signature {
        input: vec![BaseType::Bool],
        output: BaseType::Bool,
    };

    let positive_examples = vec![TestCase {
        inputs: vec![Constant::Bool(true)],
        output: Constant::Bool(false),
    }];

    let negative_examples = vec![TestCase {
        inputs: vec![Constant::Bool(false)],
        output: Constant::Bool(false),
    }];

    let prog = synthesis(
        sig,
        &bool_library(),
        Examples::new(positive_examples, negative_examples),
        3,
    );
    insta::assert_display_snapshot!("inductive_synthesis_negation", prog.unwrap());
}
