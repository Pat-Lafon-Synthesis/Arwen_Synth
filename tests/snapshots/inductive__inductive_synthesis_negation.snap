---
source: tests/inductive.rs
expression: prog.unwrap()
---
not(arg0)
